File: duplication/java/Duplication/src/main/java/repository/CustomerRepository.java
package repository;

import domain.cms.Customer;
import domain.cms.Gender;
import org.joda.time.LocalDate;
import org.joda.time.Period;
import org.joda.time.PeriodType;

public class CustomerRepository {

    // Mimic a legacy create method
    public void create(Customer customer) {
        if (customer.getGender() == Gender.UNKNOWN
                || customer.getDateOfBirth() == null) {
            throw new IllegalArgumentException(customer
                    + " not fully initialized");
        }

        LocalDate now = new LocalDate();
        Period period = new Period(customer.getDateOfBirth(),
                now, PeriodType.yearMonthDay());
        if (period.getYears() < 18) {
            throw new IllegalArgumentException(customer
                    + " is underage");
        }

        // Equally scary logic for saving would go here ...

    }

    // Mimic a legacy update method
    public void update(Customer customer) {
        if (customer.getGender() == Gender.UNKNOWN
                || customer.getDateOfBirth() == null) {
            throw new IllegalArgumentException(customer
                    + " not fully initialized");
        }

        LocalDate now = new LocalDate();
        Period period
                = new Period(customer.getDateOfBirth(),
                now, PeriodType.yearMonthDay());
        if (period.getYears() < 18) {
            throw new IllegalArgumentException(customer
                    + " is underage");
        }
    }
}

File: duplication/csharp/Duplication/Duplication/Repository/CustomerRepository.cs
using Duplication.Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Duplication.Repository
{
    public class CustomerRepository
    {

        // Mimic a legacy create method
        public void Create(Customer customer)
        {
            if (customer.Gender == Gender.Unknown || !customer.YearOfBirth.HasValue)
            {
                throw new ArgumentException("Incomplete customer");
            }

            if (DateTime.Now.Year - customer.YearOfBirth.Value < 18)
            {
                throw new ArgumentOutOfRangeException("Underage customer");
            }

            // Equally scary logic for saving would go here...

        }

        // Mimic a legacy update method
        public void Update(Customer customer)
        {
            if (customer.Gender == Gender.Unknown || !customer.YearOfBirth.HasValue)
            {
                throw new ArgumentException("Incomplete customer");
            }

            if (DateTime.Now.Year - customer.YearOfBirth.Value < 18)
            {
                throw new ArgumentOutOfRangeException("Underage customer");
            }
        }
    }

}
